src/hunk_header.rs

```diff
@@ -117,7 +117,10 @@ fn get_painted_file_with_line_number(
         )
     }
     let file_with_line_number = ansi_term::ANSIStrings(&file_with_line_number).to_string();
-    if config.hyperlinks {
+    if config.hyperlinks
+        && (config.hunk_header_style_include_file_path
+            || config.hunk_header_style_include_line_number)
+    {
```

Comment:
I wonder whether it would be slightly cleaner to test `file_with_line_number.is_empty()` here rather than re-using the two config flags, which have already been consulted above.